### PR TITLE
Add EditRequestSaver

### PR DESCRIPTION
**Pull request by toban**
- move saving of edit requests to its own service

### CODE REVIEW

File: includes/MediaWiki/Api/SingleEditEndpoint.php [hunk 1]
```diff
@@ -59,29 +50,15 @@ public function run() {
...
+		$saveStatus = $this->editRequestSaver->persistEdits( [ $request ], $this->editToken, $this->user );
 
 		// Make some sort of response
 		$response = [
 			'success' => $saveStatus->isGood()
 		];
 		if ( $saveStatus->isGood() ) {
 			/** @var EntityRevision $entityRevision */
-			$entityRevision = $saveStatus->getValue()['revision'];
+			$entityRevision = $saveStatus->getValue()[0]['revision'];
```
Comment (toban) [inline includes/MediaWiki/Api/SingleEditEndpoint.php:59]: this might need a check for that element maybe?

Comment (toban) [reply includes/MediaWiki/Api/SingleEditEndpoint.php:59]: Or maybe not, if we reach this point and there is nothing here It should crash and burn probably.

Comment (lucaswerkmeister) [reply includes/MediaWiki/Api/SingleEditEndpoint.php:59]: :+1:

File: includes/MediaWiki/Request/EditRequestSaver.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+				$request->reconcilePropertyId()
+			);
+			$otherItems = $request->otherItems();
+
+			$otherStatus = $this->persistItem( $reconciledItem, $otherItems, $editToken, $user );
+			if ( !$otherStatus->isOk() ) {
+				break;
+			}
+			$status->setResult( true, array_merge( $status->getValue(), [ $otherStatus->getValue() ] ) );
+			$status->merge( $otherStatus );
```
Comment (lucaswerkmeister) [inline includes/MediaWiki/Request/EditRequestSaver.php]: This line should move up, I think, like it did in #50?

File: includes/MediaWiki/Request/EditRequestSaver.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+		}
+
+		return $status;
+	}
+
+	/**
+	 * @param ReconciledItem $reconciledItem
+	 * @param ReconciliationServiceItem[] $otherItems
```
Comment (lucaswerkmeister) [inline includes/MediaWiki/Request/EditRequestSaver.php:64]: The `$editToken` and `$user` params seem to be missing.

File: includes/ServiceWiring.php [hunk 4]
```diff
@@ -29,6 +30,18 @@
 		);
 	},
 
+	'WikibaseReconcileEdit.EditRequestSaver' => function ( MediaWikiServices $services ): EditRequestSaver {
+		$repo = WikibaseRepo::getDefaultInstance();
+		$editEntityFactory = method_exists( $repo, 'getEditEntityFactory' )
+			? $repo->getEditEntityFactory() // 1.36+
```
Comment (lucaswerkmeister) [inline includes/ServiceWiring.php]: Hm, in the 1.36 version we could actually pass `$services` into this method, to ensure that the edit entity factory comes from the same service container. But I’m not sure if that’s very important here.

**Review by lucaswerkmeister — COMMENTED** [review]
Looks alright so far, but there are no tests for the new class yet…

File: includes/MediaWiki/Request/EditRequestSaver.php (hunk 2 above)
Comment (toban) [reply includes/MediaWiki/Request/EditRequestSaver.php]: yup, good catch, must have moved back when rebasing.

File: includes/MediaWiki/Request/EditRequestSaver.php (hunk 3 above)
Comment (toban) [reply includes/MediaWiki/Request/EditRequestSaver.php:64]: done

File: includes/ServiceWiring.php (hunk 4 above)
Comment (toban) [reply includes/ServiceWiring.php]: okay

**Review by lucaswerkmeister — APPROVED** [review]

**Pull request merged**